core/trino-main/src/main/java/io/trino/testing/LocalQueryRunner.java

```diff
@@ -691,7 +694,10 @@ public ScheduledExecutorService getScheduler()
     @Override
     public Session getDefaultSession()
     {
-        return defaultSession;
+        return TestingSession
```

Comment:
`QueryRunner.getDefaultSession` is an accessor, it should return same session each time.

Reply:
also commented about by @sopel39 here: https://github.com/trinodb/trino/pull/11955/commits/ef3beea68fc91255a8f947409f5fe8a9d27fd417#r850564469